extension.js

```diff
@@ -330,6 +375,9 @@ function unmaximize_window(win) {
 
     if (target_rect.height < workarea.height)
         win.unmaximize(Meta.MaximizeFlags.VERTICAL);
+
+    if (target_rect.width < workarea.width)
+        win.unmaximize(Meta.MaximizeFlags.HORIZONTAL);
```

Comment:
`unmaximize_window` is only called when the window is maximized vertically: `win.connect('notify::maximized-vertically', unmaximize_window);`
`win.unmaximize(Meta.MaximizeFlags.HORIZONTAL);` should be in another function, called in response to `notify::maximized-horizontally`

Reply:
renamed `unmaximize_window` to `unmaximize_window_vertically` and added `unmaximize_window_horizontally` and connected and disconnected them in appropriate places.